File: solve/0527_바탕화면 정리/postforty/바탕화면 정리.js
function solution(wallpaper) {
  const answer = [];

  let { lux, luy } = {
    lux: wallpaper.length - 1,
    luy: wallpaper[0].length - 1,
  };
  let rdx = (rdy = 0);

  wallpaper.forEach((v, x) =>
    [...v].forEach((w, y) => {
      if (w === "#") {
        if (x < lux) {
          lux = x;
        }
        if (y < luy) {
          luy = y;
        }
        if (x > rdx) {
          rdx = x;
        }
        if (y > rdy) {
          rdy = y;
        }
      }
    })
  );

  answer.push(lux);
  answer.push(luy);
  answer.push(rdx + 1);
  answer.push(rdy + 1);

  return answer;
}

// console.log(solution([".#...", "..#..", "...#."])); // [0, 1, 3, 4]
// console.log(solution(["..", "#."])); // [1, 0, 2, 1]
console.log(solution(["...#"]));

File: solve/0527_바탕화면 정리/postforty/바탕화면 정리.py
def solution(wallpaper):
    lux, luy = (len(wallpaper) - 1, len(wallpaper[0]) - 1)
    rdx = rdy = 0

    for x, v in enumerate(wallpaper):
        for y, w in enumerate(v):
            if w == '#':
                if x < lux:
                    lux = x
                if y < luy:
                    luy = y
                if x > rdx:
                    rdx = x
                if y > rdy:
                    rdy = y

    return [lux, luy, rdx + 1, rdy + 1]

# print(solution([".#...", "..#..", "...#."])) # [0, 1, 3, 4]
# print(solution(["..", "#."])) # [1, 0, 2, 1]
print(solution(["...#"])) # [0, 3, 1, 4]